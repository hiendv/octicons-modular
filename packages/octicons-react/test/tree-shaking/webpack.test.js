/**
 * @jest-environment node
 */

import path from 'path'
import webpack from 'webpack'
import MemoryFS from 'memory-fs'

describe('Webpack tree-shaking', () => {
  test('works', () => {
    const fs = new MemoryFS()
    const compiler = webpack({
      entry: path.resolve(__dirname, 'main.js'),
      output: {
        filename: 'main.js',
        path: '/'
      },
      optimization: {
        sideEffects: true
      },
      target: [ 'web', 'es5' ],
      mode: 'production'
    })

    compiler.outputFileSystem = fs

    return new Promise((resolve) => {
      compiler.run((err, stats) => {
        resolve({ err, stats })
      })
    }).then(({ err, stats }) => {
      expect(err).toBeFalsy()
      expect(stats.hasErrors()).toBeFalsy()

      const code = fs.readFileSync('/main.js').toString('utf8')
      expect(code).toMatchSnapshot()
    })
  })
})
