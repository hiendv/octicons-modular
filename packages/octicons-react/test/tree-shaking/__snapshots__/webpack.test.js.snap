// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Webpack tree-shaking works 1`] = `
"!function(e){var t={};function r(n){if(t[n])return t[n].exports;var o=t[n]={i:n,l:!1,exports:{}};return e[n].call(o.exports,o,o.exports,r),o.l=!0,o.exports}r.m=e,r.c=t,r.d=function(e,t,n){r.o(e,t)||Object.defineProperty(e,t,{enumerable:!0,get:n})},r.r=function(e){\\"undefined\\"!=typeof Symbol&&Symbol.toStringTag&&Object.defineProperty(e,Symbol.toStringTag,{value:\\"Module\\"}),Object.defineProperty(e,\\"__esModule\\",{value:!0})},r.t=function(e,t){if(1&t&&(e=r(e)),8&t)return e;if(4&t&&\\"object\\"==typeof e&&e&&e.__esModule)return e;var n=Object.create(null);if(r.r(n),Object.defineProperty(n,\\"default\\",{enumerable:!0,value:e}),2&t&&\\"string\\"!=typeof e)for(var o in e)r.d(n,o,function(t){return e[t]}.bind(null,o));return n},r.n=function(e){var t=e&&e.__esModule?function(){return e.default}:function(){return e};return r.d(t,\\"a\\",t),t},r.o=function(e,t){return Object.prototype.hasOwnProperty.call(e,t)},r.p=\\"\\",r(r.s=189)}({0:function(e,t,r){e.exports=r(185)()},1:function(e,t,r){\\"use strict\\";e.exports=r(187)},185:function(e,t,r){\\"use strict\\";var n=r(186);function o(){}function u(){}u.resetWarningCache=o,e.exports=function(){function e(e,t,r,o,u,c){if(c!==n){var a=new Error(\\"Calling PropTypes validators directly is not supported by the \`prop-types\` package. Use PropTypes.checkPropTypes() to call them. Read more at http://fb.me/use-check-prop-types\\");throw a.name=\\"Invariant Violation\\",a}}function t(){return e}e.isRequired=e;var r={array:e,bool:e,func:e,number:e,object:e,string:e,symbol:e,any:e,arrayOf:t,element:e,elementType:e,instanceOf:t,node:e,objectOf:t,oneOf:t,oneOfType:t,shape:t,exact:t,checkPropTypes:u,resetWarningCache:o};return r.PropTypes=r,r}},186:function(e,t,r){\\"use strict\\";e.exports=\\"SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED\\"},187:function(e,t,r){\\"use strict\\";
/** @license React v16.9.0
 * react.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */var n=r(188),o=\\"function\\"==typeof Symbol&&Symbol.for,u=o?Symbol.for(\\"react.element\\"):60103,c=o?Symbol.for(\\"react.portal\\"):60106,a=o?Symbol.for(\\"react.fragment\\"):60107,i=o?Symbol.for(\\"react.strict_mode\\"):60108,l=o?Symbol.for(\\"react.profiler\\"):60114,f=o?Symbol.for(\\"react.provider\\"):60109,s=o?Symbol.for(\\"react.context\\"):60110,p=o?Symbol.for(\\"react.forward_ref\\"):60112,y=o?Symbol.for(\\"react.suspense\\"):60113,d=o?Symbol.for(\\"react.suspense_list\\"):60120,h=o?Symbol.for(\\"react.memo\\"):60115,v=o?Symbol.for(\\"react.lazy\\"):60116;o&&Symbol.for(\\"react.fundamental\\"),o&&Symbol.for(\\"react.responder\\");var m=\\"function\\"==typeof Symbol&&Symbol.iterator;function b(e){for(var t=e.message,r=\\"https://reactjs.org/docs/error-decoder.html?invariant=\\"+t,n=1;n<arguments.length;n++)r+=\\"&args[]=\\"+encodeURIComponent(arguments[n]);return e.message=\\"Minified React error #\\"+t+\\"; visit \\"+r+\\" for the full message or use the non-minified dev environment for full errors and additional helpful warnings. \\",e}var g={isMounted:function(){return!1},enqueueForceUpdate:function(){},enqueueReplaceState:function(){},enqueueSetState:function(){}},_={};function S(e,t,r){this.props=e,this.context=t,this.refs=_,this.updater=r||g}function O(){}function j(e,t,r){this.props=e,this.context=t,this.refs=_,this.updater=r||g}S.prototype.isReactComponent={},S.prototype.setState=function(e,t){if(\\"object\\"!=typeof e&&\\"function\\"!=typeof e&&null!=e)throw b(Error(85));this.updater.enqueueSetState(this,e,t,\\"setState\\")},S.prototype.forceUpdate=function(e){this.updater.enqueueForceUpdate(this,e,\\"forceUpdate\\")},O.prototype=S.prototype;var w=j.prototype=new O;w.constructor=j,n(w,S.prototype),w.isPureReactComponent=!0;var k={current:null},x={suspense:null},C={current:null},E=Object.prototype.hasOwnProperty,P={key:!0,ref:!0,__self:!0,__source:!0};function R(e,t,r){var n=void 0,o={},c=null,a=null;if(null!=t)for(n in void 0!==t.ref&&(a=t.ref),void 0!==t.key&&(c=\\"\\"+t.key),t)E.call(t,n)&&!P.hasOwnProperty(n)&&(o[n]=t[n]);var i=arguments.length-2;if(1===i)o.children=r;else if(1<i){for(var l=Array(i),f=0;f<i;f++)l[f]=arguments[f+2];o.children=l}if(e&&e.defaultProps)for(n in i=e.defaultProps)void 0===o[n]&&(o[n]=i[n]);return{$$typeof:u,type:e,key:c,ref:a,props:o,_owner:C.current}}function $(e){return\\"object\\"==typeof e&&null!==e&&e.$$typeof===u}var T=/\\\\/+/g,I=[];function N(e,t,r,n){if(I.length){var o=I.pop();return o.result=e,o.keyPrefix=t,o.func=r,o.context=n,o.count=0,o}return{result:e,keyPrefix:t,func:r,context:n,count:0}}function A(e){e.result=null,e.keyPrefix=null,e.func=null,e.context=null,e.count=0,10>I.length&&I.push(e)}function L(e,t,r){return null==e?0:function e(t,r,n,o){var a=typeof t;\\"undefined\\"!==a&&\\"boolean\\"!==a||(t=null);var i=!1;if(null===t)i=!0;else switch(a){case\\"string\\":case\\"number\\":i=!0;break;case\\"object\\":switch(t.$$typeof){case u:case c:i=!0}}if(i)return n(o,t,\\"\\"===r?\\".\\"+q(t,0):r),1;if(i=0,r=\\"\\"===r?\\".\\":r+\\":\\",Array.isArray(t))for(var l=0;l<t.length;l++){var f=r+q(a=t[l],l);i+=e(a,f,n,o)}else if(null===t||\\"object\\"!=typeof t?f=null:f=\\"function\\"==typeof(f=m&&t[m]||t[\\"@@iterator\\"])?f:null,\\"function\\"==typeof f)for(t=f.call(t),l=0;!(a=t.next()).done;)i+=e(a=a.value,f=r+q(a,l++),n,o);else if(\\"object\\"===a)throw n=\\"\\"+t,b(Error(31),\\"[object Object]\\"===n?\\"object with keys {\\"+Object.keys(t).join(\\", \\")+\\"}\\":n,\\"\\");return i}(e,\\"\\",t,r)}function q(e,t){return\\"object\\"==typeof e&&null!==e&&null!=e.key?function(e){var t={\\"=\\":\\"=0\\",\\":\\":\\"=2\\"};return\\"$\\"+(\\"\\"+e).replace(/[=:]/g,(function(e){return t[e]}))}(e.key):t.toString(36)}function M(e,t){e.func.call(e.context,t,e.count++)}function U(e,t,r){var n=e.result,o=e.keyPrefix;e=e.func.call(e.context,t,e.count++),Array.isArray(e)?F(e,n,r,(function(e){return e})):null!=e&&($(e)&&(e=function(e,t){return{$$typeof:u,type:e.type,key:t,ref:e.ref,props:e.props,_owner:e._owner}}(e,o+(!e.key||t&&t.key===e.key?\\"\\":(\\"\\"+e.key).replace(T,\\"$&/\\")+\\"/\\")+r)),n.push(e))}function F(e,t,r,n,o){var u=\\"\\";null!=r&&(u=(\\"\\"+r).replace(T,\\"$&/\\")+\\"/\\"),L(e,U,t=N(t,u,n,o)),A(t)}function B(){var e=k.current;if(null===e)throw b(Error(321));return e}var z={Children:{map:function(e,t,r){if(null==e)return e;var n=[];return F(e,n,null,t,r),n},forEach:function(e,t,r){if(null==e)return e;L(e,M,t=N(null,null,t,r)),A(t)},count:function(e){return L(e,(function(){return null}),null)},toArray:function(e){var t=[];return F(e,t,null,(function(e){return e})),t},only:function(e){if(!$(e))throw b(Error(143));return e}},createRef:function(){return{current:null}},Component:S,PureComponent:j,createContext:function(e,t){return void 0===t&&(t=null),(e={$$typeof:s,_calculateChangedBits:t,_currentValue:e,_currentValue2:e,_threadCount:0,Provider:null,Consumer:null}).Provider={$$typeof:f,_context:e},e.Consumer=e},forwardRef:function(e){return{$$typeof:p,render:e}},lazy:function(e){return{$$typeof:v,_ctor:e,_status:-1,_result:null}},memo:function(e,t){return{$$typeof:h,type:e,compare:void 0===t?null:t}},useCallback:function(e,t){return B().useCallback(e,t)},useContext:function(e,t){return B().useContext(e,t)},useEffect:function(e,t){return B().useEffect(e,t)},useImperativeHandle:function(e,t,r){return B().useImperativeHandle(e,t,r)},useDebugValue:function(){},useLayoutEffect:function(e,t){return B().useLayoutEffect(e,t)},useMemo:function(e,t){return B().useMemo(e,t)},useReducer:function(e,t,r){return B().useReducer(e,t,r)},useRef:function(e){return B().useRef(e)},useState:function(e){return B().useState(e)},Fragment:a,Profiler:l,StrictMode:i,Suspense:y,unstable_SuspenseList:d,createElement:R,cloneElement:function(e,t,r){if(null==e)throw b(Error(267),e);var o=void 0,c=n({},e.props),a=e.key,i=e.ref,l=e._owner;if(null!=t){void 0!==t.ref&&(i=t.ref,l=C.current),void 0!==t.key&&(a=\\"\\"+t.key);var f=void 0;for(o in e.type&&e.type.defaultProps&&(f=e.type.defaultProps),t)E.call(t,o)&&!P.hasOwnProperty(o)&&(c[o]=void 0===t[o]&&void 0!==f?f[o]:t[o])}if(1===(o=arguments.length-2))c.children=r;else if(1<o){f=Array(o);for(var s=0;s<o;s++)f[s]=arguments[s+2];c.children=f}return{$$typeof:u,type:e.type,key:a,ref:i,props:c,_owner:l}},createFactory:function(e){var t=R.bind(null,e);return t.type=e,t},isValidElement:$,version:\\"16.9.0\\",unstable_withSuspenseConfig:function(e,t){var r=x.suspense;x.suspense=void 0===t?null:t;try{e()}finally{x.suspense=r}},__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED:{ReactCurrentDispatcher:k,ReactCurrentBatchConfig:x,ReactCurrentOwner:C,IsSomeRendererActing:{current:!1},assign:n}},D={default:z},V=D&&z||D;e.exports=V.default||V},188:function(e,t,r){\\"use strict\\";
/*
object-assign
(c) Sindre Sorhus
@license MIT
*/var n=Object.getOwnPropertySymbols,o=Object.prototype.hasOwnProperty,u=Object.prototype.propertyIsEnumerable;function c(e){if(null==e)throw new TypeError(\\"Object.assign cannot be called with null or undefined\\");return Object(e)}e.exports=function(){try{if(!Object.assign)return!1;var e=new String(\\"abc\\");if(e[5]=\\"de\\",\\"5\\"===Object.getOwnPropertyNames(e)[0])return!1;for(var t={},r=0;r<10;r++)t[\\"_\\"+String.fromCharCode(r)]=r;if(\\"0123456789\\"!==Object.getOwnPropertyNames(t).map((function(e){return t[e]})).join(\\"\\"))return!1;var n={};return\\"abcdefghijklmnopqrst\\".split(\\"\\").forEach((function(e){n[e]=e})),\\"abcdefghijklmnopqrst\\"===Object.keys(Object.assign({},n)).join(\\"\\")}catch(e){return!1}}()?Object.assign:function(e,t){for(var r,a,i=c(e),l=1;l<arguments.length;l++){for(var f in r=Object(arguments[l]))o.call(r,f)&&(i[f]=r[f]);if(n){a=n(r);for(var s=0;s<a.length;s++)u.call(r,a[s])&&(i[a[s]]=r[a[s]])}}return i}},189:function(e,t,r){\\"use strict\\";r.r(t);var n=r(0),o=r.n(n),u=r(1),c=r.n(u),a=function(e){var t=e.icon,r={scale:e.scale,class:e.className,label:e.label},n=t.attrs(r);if(!n)return null;Object.keys(n).forEach((function(e){n[e]=\\"\\"+n[e]})),n.className=n.class,delete n.class;var o=t.path(!0);return o.key=\\"p0\\",c.a.createElement(\\"svg\\",n,[c.a.createElement(\\"path\\",o)])};a.propTypes={icon:o.a.shape({attrs:o.a.func.isRequired,path:o.a.func.isRequired}),scale:o.a.number,className:o.a.string,label:o.a.string},a.defaultProps={scale:1,className:null,label:null};var i=a,l=function(e){for(var t=[],r=arguments.length-1;r-- >0;)t[r]=arguments[r+1];for(var n,o=0,u=t.length;o<u;o++)n=t[o],Object.keys(n).forEach((function(t){\\"__proto__\\"!==t&&(e[t]=n[t])}));return e},f=function(e){return e.toLowerCase().replace(/[^a-zA-Z0-9]+(.)/g,(function(e,t){return t.toUpperCase()}))};var s=function(e,t,r,n,o){var u=function(t,r){r.label?t[\\"aria-label\\"]=r.label:t[\\"aria-hidden\\"]=!0,r.class?t.class=\\"octicon octicon-\\"+e+\\" \\"+r.class:t.class=\\"octicon octicon-\\"+e;var n=0===r.scale?0:parseFloat(r.scale)||1,o=n*parseInt(t.width),u=n*parseInt(t.height);return t.width=Number(o.toFixed(2)),t.height=Number(u.toFixed(2)),t},c=function(e){return Object.keys(e).map((function(t){return t+'=\\"'+e[t]+'\\"'})).join(\\" \\").trim()};return{name:e,path:function(e){void 0===e&&(e=!1);var t=l({},n);if(!e)return t;var r={};return Object.keys(t).forEach((function(e){r[f(e)]=t[e]})),r},keywords:function(){return o},attrs:function(e){return function(e){var n=l({scale:1,label:null,class:null},e||{});return u({version:\\"1.1\\",width:t,height:r,viewBox:\\"0 0 \\"+t+\\" \\"+r},n)}(e)},html:function(e){return\\"<svg \\"+c(this.attrs(e))+\\"><path \\"+c(this.path())+\\"/></svg>\\"}}}(\\"alert\\",16,16,{\\"fill-rule\\":\\"evenodd\\",d:\\"M8.893 1.5c-.183-.31-.52-.5-.887-.5s-.703.19-.886.5L.138 13.499a.98.98 0 0 0 0 1.001c.193.31.53.501.886.501h13.964c.367 0 .704-.19.877-.5a1.03 1.03 0 0 0 .01-1.002L8.893 1.5zm.133 11.497H6.987v-2.003h2.039v2.003zm0-3.004H6.987V5.987h2.039v4.006z\\"},[\\"warning\\",\\"triangle\\",\\"exclamation\\",\\"point\\"]);!function(e,t){void 0===t&&(t={});var r=t.insertAt;if(e&&\\"undefined\\"!=typeof document){var n=document.head||document.getElementsByTagName(\\"head\\")[0],o=document.createElement(\\"style\\");o.type=\\"text/css\\",\\"top\\"===r&&n.firstChild?n.insertBefore(o,n.firstChild):n.appendChild(o),o.styleSheet?o.styleSheet.cssText=e:o.appendChild(document.createTextNode(e))}}(\\".octicon{display:inline-block;vertical-align:text-top;fill:currentColor}\\");var p=i;console.log(p,s.svg())}});"
`;
